src/pandroid/app/src/main/java/com/panda3ds/pandroid/app/MainActivity.java

```diff
@@ -0,0 +1,58 @@
+package com.panda3ds.pandroid.app;
+
+import static android.Manifest.permission.READ_EXTERNAL_STORAGE;
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.provider.Settings.ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION;
+
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Environment;
+import android.widget.Toast;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+import com.panda3ds.pandroid.R;
+import com.panda3ds.pandroid.utils.Constants;
+import com.panda3ds.pandroid.utils.PathUtils;
+
+public class MainActivity extends BaseActivity {
+	private static final int PICK_3DS_ROM = 2;
+	private static final int PERMISSION_REQUEST_CODE = 3;
+
+	private void openFile() {
+		Intent intent = new Intent(Intent.ACTION_OPEN_DOCUMENT);
+		intent.addCategory(Intent.CATEGORY_OPENABLE);
+		intent.setType("*/*");
```

Comment:
Can this be used as a file filter or does it only accept officially recognized MIME types?